packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -199,18 +199,51 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
 
     if (!kReleaseMode) {
       int frameNumber = 0;
-
-      // use frameTimings. https://github.com/flutter/flutter/issues/38838
-      // ignore: deprecated_member_use
-      window.onReportTimings = (List<FrameTiming> timings) {
+      addTimingsCallback((List<FrameTiming> timings) {
         for (FrameTiming frameTiming in timings) {
           frameNumber += 1;
           _profileFramePostEvent(frameNumber, frameTiming);
         }
+      });
+    }
+  }
+
+  /// Add a [TimingsCallback] that receives [FrameTiming] sent from the engine.
+  ///
+  /// This can be used, for example, to monitor the performance in release mode,
+  /// or to get a signal when the first frame is rasterized.
+  void addTimingsCallback(TimingsCallback callback) {
+    // TODO(liyuqian): once this is merged, modify the doc of
+    //  [Window.onReportTimings] inside the engine repo to recommend using this
+    // API instead of using [Window.onReportTimings] directly.
+    _timingsCallbacksEnabled[callback] = true;
+    if (_timingsCallbacks.length == 1) {
+      assert(window.onReportTimings == null);
+      window.onReportTimings = (List<FrameTiming> timings) {
+        for (TimingsCallback callback in _timingsCallbacks) {
+          callback(timings);
```

Comment:
What if callback removes itself from _timingsCallbacks? You probably want to iterate over a copy of _timingsCallbacks to avoid problems (_timingsCallbacks.toList() should work).